subsys/net/lib/app/net_app.c

```diff
@@ -393,27 +440,58 @@ struct net_context *_net_app_select_net_ctx(struct net_app_ctx *ctx,
 {
 #if defined(CONFIG_NET_APP_CLIENT)
 	if (ctx->app_type == NET_APP_CLIENT) {
```

Comment:
Can this function be split into at least two, one selecting the context for servers, and other for clients? It's quite large, full of conditional compilation, and deeply nested.